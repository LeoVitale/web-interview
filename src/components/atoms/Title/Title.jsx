import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'components/atoms/Icon'
import { labelSpan, titleIcon } from './Title.module.scss'

const Title = ({ icon, label }) => {
  return (
    <h3>
      {icon && <Icon className={titleIcon} type={icon} />}
      <span className={labelSpan}>{label}</span>
    </h3>
  )
}

Title.propTypes = {
  icon: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
}

export default Title
