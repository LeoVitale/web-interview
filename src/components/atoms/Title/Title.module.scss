.labelSpan {
  vertical-align: middle;
}

.titleIcon {
  margin-right: 10px;
}
