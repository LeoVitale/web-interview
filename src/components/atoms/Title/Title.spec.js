import React from 'react'
import { mount } from 'enzyme'
import Title from './Title.jsx'

describe('<Title />', () => {
  const title = mount(<Title />)

  it('should render Title as expected', () => {
    expect(title).toMatchSnapshot()
  })
})
