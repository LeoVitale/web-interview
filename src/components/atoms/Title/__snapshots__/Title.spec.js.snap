// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render Title as expected 1`] = `
<Title>
  <h3>
    <span
      className="labelSpan"
    />
  </h3>
</Title>
`;
